test/Spec.hs

```diff
@@ -1,14 +1,40 @@
 import Boids
+import Utils (dosedLists, saltedRange)
 import Test.HUnit
 import Test.HUnit.Approx
 import System.Exit
 
+
 testRadsToDegrees :: Test
 testRadsToDegrees = TestCase (assertApproxEqual "Rads to Degrees" 1e-6 180.0 (radsToDegrees pi))
 
+
+testDosedLists :: Test
+testDosedLists = TestCase $ do
+  assertEqual "Dosed by 3 elements" [[1,2,3],[4,5,6],[7,8,9]]
+                                    (dosedLists 3 [1..9])
+  assertEqual "Dosed by 2 elements" [[1,2],[3,4],[5,6],[7,8]]
+                                    (dosedLists 2 [1..8])
+  assertEqual "Dosed by 1 element" [[1],[2],[3],[4],[5]]
+                                   (dosedLists 1 [1..5])
+  assertEqual "Small tails are sliced" [[1,2,3],[4,5,6],[7,8,9]]
+                                       (dosedLists 3 [1..10])
+  assertEqual "Works well with infinite lists" [[1,2,3],[4,5,6],[7,8,9]]
+                                               (take 3 $ dosedLists 3 [1..])
+
+testSaltedRange :: Test
+testSaltedRange = TestCase $ do
+  assertEqual "Middle" 30 (saltedRange 0.5 (20, 40))
+  assertEqual "Minimum" 20 (saltedRange 0 (20, 40))
+  assertEqual "Maximum" 40 (saltedRange 1 (20, 40))
+
+
 main :: IO Counts
-main = do results <- runTestTT $ TestList [TestLabel "Test Radians to Degrees conversion" testRadsToDegrees]
-          if (errors results + failures results == 0)
-          then exitWith ExitSuccess
+main = do results <- runTestTT tests
+          if errors results + failures results == 0
+          then exitSuccess
           else exitWith (ExitFailure 1)
```

Comment:
Please, use `exitFailure` here as well. Docs say that 

> The computation `exitFailure` is equivalent to `exitWith (ExitFailure exitfail)`, where `exitfail` is implementation-dependent

The exact exit code doesn't actually matter here. I just picked `1` because I didn't know about `exitSuccess` and `exitFailure`.

So my I assumption here is if it's implementation-dependent it will be different depending on the exact implementation  of a Haskell compiler for a specific platform. Which effectively makes it more crossplatform. Please, correct me if I'm wrong.